File: leetcode/014-longest-common-prefix.cpp
class Solution
{
 public:
  string longestCommonPrefix(vector<string>& strs)
  {
    if (strs.empty()) return "";

    string ret;
    for (int i = 0; /* empty */; ++i) {
      if (strs[0].size() == i) break;
      const char c = strs[0][i];
      for (int j = 1; j < strs.size(); ++j) {
        if (strs[j].size() == i || c != strs[j][i]) {
          i = -1;
          break;
        }
      }

      if (i >= 0) ret += c;
      else break;
    }
    return ret;
  }
};

File: leetcode/014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ""

        ret = ""
        i = 0
        while True:
            if len(strs[0]) == i:
                break

            c = strs[0][i]
            for s in strs[1:]:
                if len(s) == i or s[i] != c:
                    i = -1
                    break

            if i >= 0: ret += c
            else: break

            i += 1

        return ret
